### ISSUE TITLE
Update Groups  via zbx.hosts.update

### ISSUE THREAD

**Issue by sfrostick** [express42/zabbixapi#31]
This is what i'm trying to run 

``` 
zbx.hosts.update(
            :hostid  =>  zbx.hosts.get_id(:host => "hostname"),
            :groups => [ :groupid => 12],

        )
```

The command completes silently but the group is not updated. I get the following when debug is true

```
[DEBUG] Equal keys {:hostid=>"10136", :proxy_hostid=>"0", :host=>"hostname", :status=>"0", :disable_until=>"0", :error=>"", :available=>"1", :errors_from=>"0", :lastaccess=>"0", :ipmi_authtype=>"0", :ipmi_privilege=>"2", :ipmi_username=>"", :ipmi_password=>"", :ipmi_disable_until=>"0", :ipmi_available=>"0", :snmp_disable_until=>"0", :snmp_available=>"0", :maintenanceid=>"0", :maintenance_status=>"0", :maintenance_type=>"0", :maintenance_from=>"0", :ipmi_errors_from=>"0", :snmp_errors_from=>"0", :ipmi_error=>"", :snmp_error=>"", :jmx_disable_until=>"0", :jmx_available=>"0", :jmx_errors_from=>"0", :jmx_error=>"", :name=>"hostname", :flags=>"0", :templateid=>"0", :description=>""} and {:hostid=>10136, :groups=>[{:groupid=>12}]}, skip update
```

It seems the current code can't handle a change of group, looks like there is two places which are causing problems, firstly in the normalize_hash function the array is deleted.

```
 def normalize_hash(hash)
      result = hash.dup
      result.delete(:hostid) #TODO remove to logig. TemplateID and HostID has different id 
      result.each do |key, value|
        case value
          when Array
            result.delete(key)
          else
            result[key] = value.to_s
        end
      end
      result
    end

```

Also the code never retrieves a list of host groups to compare to the only data pulled down is

```
Send request: {"method":"host.get","params":{"filter":{"host":"hostname"},"output":"extend"},"id":16595,"jsonrpc":"2.0","auth":"ea76bb5ebf32728ab518af7daef6bd55"}
```
```
Get answer: {"jsonrpc":"2.0","result":[{"hostid":"10136","proxy_hostid":"0","host":"hostname","status":"0","disable_until":"0","error":"","available":"1","errors_from":"0","lastaccess":"0","ipmi_authtype":"0","ipmi_privilege":"2","ipmi_username":"","ipmi_password":"","ipmi_disable_until":"0","ipmi_available":"0","snmp_disable_until":"0","snmp_available":"0","maintenanceid":"0","maintenance_status":"0","maintenance_type":"0","maintenance_from":"0","ipmi_errors_from":"0","snmp_errors_from":"0","ipmi_error":"","snmp_error":"","jmx_disable_until":"0","jmx_available":"0","jmx_errors_from":"0","jmx_error":"","name":"hostname","flags":"0","templateid":"0","description":""}],"id":16595}
```

This means that when the hashes are compared the keys are equal and the code failed to update.
Something like this does work though when you define a variable that is retrieved in the host.get call

``` 
zbx.hosts.update(
            :hostid  =>  zbx.hosts.get_id(:host => "hostname"),
            :groups => [ :groupid => 12],
            :name => "test"

        )
```

**Comment by sfrostick** [comment]
A work around for the moment for us as we don't use jmx is to generate a random jmx_available value this will allow the groups to be updated each time we run.

```
 zbx.hosts.update(
            :hostid  =>  zbx.hosts.get_id(:host => host),
            :groups => [ :groupid => hostgroupid ],
            :jmx_available => r.rand(1..100).to_s
        )
```

**Comment by evtuhovich** [comment]
@sfrostick fix in 2.4.3, if you use older version (< 2.4) of zabbix, please notify me, i'll release other brunches too

**Comment by sfrostick** [comment]
Thanks a lot, we run zabbix 2.4 only.
